src/box/txn.c

```diff
@@ -1069,6 +1069,11 @@ txn_prepare(struct txn *txn)
 			return -1;
 		}
 	}
+	/*
+	 * Since every transaction is registered in memtx_tx (see `txn_begin`),
+	 * every transaction should finalize its prepare there as well.
+	 */
+	memtx_tx_prepare_finalize(txn);
```

Comment:
The new comment looks rather confusing because `memtx_tx_prepare_finalize` has nothing to do with `memtx_tx_register_txn` AFAICS.

Reply:
Updated the comment:
```
/*
  * Non-memtx transactions can use memtx MVCC (for example, transactions
  * reading from sysview), so we need to finalize prepare of all
  * transactions, not only memtx ones.
  */
```